cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -258,8 +275,8 @@ Result<std::vector<int>> InferColumnProjection(const parquet::arrow::FileReader&
 
   std::vector<int> columns_selection;
   for (const auto& ref : field_refs) {
-    RETURN_NOT_OK(ResolveOneFieldRef(manifest, ref, field_lookup, duplicate_fields,
-                                     &columns_selection));
+    RETURN_NOT_OK(ResolveOneFieldRef(manifest, dataset_schema, ref, field_lookup,
```

Comment:
nit, but I don't see a reason to create a copy of the dataset schema above, just pass it as `options.dataset_schema`